extensions/amp-mraid/0.1/amp-mraid.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Connects AMP Host Services to MRAID to allow invoking native
+ * APIs from mobile app ad webviews.
+ *
+ * Example:
+ * <code>
+ * <script async host-api="amp-mraid"
+ *  fallback-on="load incomplete"></script>
+ * </code>
+ *
+ */
+
+import {HostServices} from '../../../src/inabox/host-services';
+import {dev} from '../../../src/log';
+import {getMode} from '../../../src/mode';
+import {MraidService} from './mraid-service';
+
+const TAG = 'amp-mraid';
+const FALLBACK_ON = 'fallback-on';
+
+/**
+ * Error codes that can be macroed into the error ping url.
+ * @const @enum {string}
+ */
+const ErrorCodes = {
```

Comment:
we will need this errorCode part of the HostAPI. could you move it to host-services.js
also, the design was 

```
HostApiError: {
  UNKNOWN: 0,
  MISS_MATCH: 1, // the API impl mis-match the environment
  NOT_SUPPORTED: 2 // the feature is not supported
}
```

but feel free to rename

Reply:
done